tools/mo/openvino/tools/mo/middle/BlockLSTMtoLSTMSequence.py

```diff
@@ -3,13 +3,111 @@
 
 import numpy as np
 
-from openvino.tools.mo.ops.LSTM import LSTM
+from openvino.tools.mo.front.common.partial_infer.utils import dynamic_dimension
 from openvino.tools.mo.front.common.partial_infer.utils import int64_array
 from openvino.tools.mo.graph.graph import Graph
 from openvino.tools.mo.middle.replacement import MiddleReplacementPattern
+from openvino.tools.mo.ops.LSTM import LSTM
 from openvino.tools.mo.utils.error import Error
 
 
+class BlockLSTMtoLSTMSequenceSingleFirstOutput(MiddleReplacementPattern):
+    """
+    This transformation handles BlockLSTM with just one output, concatenation of all the intermediate
+    output values of the hidden.
+    TODO: implement for non-constant weights and bias.
+     For this, it requires to unbound from LSTMRNNSequenceToTensorIterator transformation
+    """
+    enabled = True
+
+    def run_before(self):
+        from openvino.tools.mo.middle.LSTMRNNSequenceToTensorIterator import LSTMToTensorIterator
+        return [LSTMToTensorIterator]
+
+    def run_after(self):
+        from openvino.tools.mo.middle.pass_separator import MiddleStart
+        from openvino.tools.mo.middle.RNNSequenceNormalizeToIE import RNNSequenceNormalize
+        return [MiddleStart, RNNSequenceNormalize, BlockLSTMtoLSTMSequence]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('BlockLSTM', dict(op='BlockLSTM')),
+                ('weights', dict(op='Const')),
+                ('weights_data', dict(kind='data')),
+                ('bias', dict(op='Const')),
+                ('bias_data', dict(kind='data')),
+            ],
+            edges=[
+                ('weights', 'weights_data'),
+                ('weights_data', 'BlockLSTM', {'in': 1}),
+                ('bias', 'bias_data'),
+                ('bias_data', 'BlockLSTM', {'in': 2}),
+            ]
+        )
+
+    @staticmethod
+    def replace_pattern(graph: Graph, match: dict):
+        block_lstm = match['BlockLSTM']
+        connected_out_ports = [port_idx for port_idx, port in block_lstm.out_ports().items() if
+                               not port.disconnected()]
+        # support only BlockLSTM with one output of concatenated hidden states from all steps
+        if len(connected_out_ports) != 1 or connected_out_ports[0] != 0:
+            return
+        shift_const = block_lstm.forget_bias
+
+        # determine hidden_size based on weights shapes
+        w_shape = block_lstm.in_port(1).data.get_shape()
+        b_shape = block_lstm.in_port(2).data.get_shape()
+        hidden_size = dynamic_dimension
+        if len(b_shape) > 0 and b_shape[0] is not dynamic_dimension:
+            hidden_size = b_shape[0] // 4
+        elif len(w_shape) > 1 and w_shape[1] is not dynamic_dimension:
+            hidden_size = w_shape[1] // 4
+
+        assert hidden_size is not dynamic_dimension
```

Comment:
Please either add comment to this assert so that is was printed in log if something went wrong, or just skip the transformation if we met dynamic shapes.